### PR TITLE
[mlir][nvgpu] Make `phaseParity` of `mbarrier.try_wait` `i1`

### PR DESCRIPTION
**Pull request by grypp**
Currently, `phaseParity` argument of `nvgpu.mbarrier.try_wait.parity` is index. This can cause a problem if it's passed any value different than 0 or 1. Because the PTX instruction only accepts even or odd phase. This PR makes phaseParity argument i1 to avoid misuse.

Here is the information from PTX doc:

```
The .parity variant of the instructions test for the completion of the phase indicated 
by the operand phaseParity, which is the integer parity of either the current phase or 
the immediately preceding phase of the mbarrier object. An even phase has integer 
parity 0 and an odd phase has integer parity of 1. So the valid values of phaseParity 
operand are 0 and 1.
```
See for more information:
https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#parallel-synchronization-and-communication-instructions-mbarrier-test-wait-mbarrier-try-wait

### CODE REVIEW

**Comment by grypp** [comment]
@durga4github can you please review this one?

File: mlir/lib/Dialect/NVGPU/IR/NVGPUDialect.cpp [hunk 1]
```diff
@@ -362,7 +362,8 @@ std::optional<InFlightDiagnostic> verifyTmaDescriptorWithMemref(
                              << kMaxTMADimension << " but it is " << dim;
     }
   }
-  if (descMemref.getRank() > 1) {
+  if (descMemref.getRank() > 1 &&
+      descType.getSwizzle() != TensorMapSwizzleKind::SWIZZLE_NONE) {
```
Comment (durga4github) [inline mlir/lib/Dialect/NVGPU/IR/NVGPUDialect.cpp:366]: Is this change intended to be part of this patch ?

File: mlir/test/Integration/GPU/CUDA/sm90/gemm_f32_f16_f16_128x128x128.mlir [hunk 2]
```diff
@@ -197,7 +197,8 @@ func.func @main() {
     {
       %ticks = arith.constant 10000000 : index
       // TMA wait
-      nvgpu.mbarrier.try_wait.parity %barrier[%i], %c0, %ticks : !barrierType
+      %phase = arith.constant 0 : i1
```
Comment (durga4github) [inline mlir/test/Integration/GPU/CUDA/sm90/gemm_f32_f16_f16_128x128x128.mlir:200]: I like the c0 kind of naming that we usually use for constants.
Shall we do %phase_c0 for this one ?
(This is optional)

**Review by durga4github — APPROVED** [review]
Looks good to me except for an optional naming change

File: mlir/test/Integration/GPU/CUDA/sm90/gemm_f32_f16_f16_128x128x128.mlir (hunk 2 above)
Comment (grypp) [reply mlir/test/Integration/GPU/CUDA/sm90/gemm_f32_f16_f16_128x128x128.mlir]: fixed

File: mlir/lib/Dialect/NVGPU/IR/NVGPUDialect.cpp (hunk 1 above)
Comment (grypp) [reply mlir/lib/Dialect/NVGPU/IR/NVGPUDialect.cpp]: It is not :) it was another fix, and leaked here. 
I put another PR for this fix
https://github.com/llvm/llvm-project/pull/81499

File: mlir/test/Integration/GPU/CUDA/sm90/gemm_pred_f32_f16_f16_128x128x128.mlir [hunk 3]
```diff
@@ -206,7 +206,8 @@ func.func @main() {
     {
       %ticks = arith.constant 10000000 : index
       // TMA wait
-      nvgpu.mbarrier.try_wait.parity %barrier[%i], %c0, %ticks : !barrierType
+      %phase = arith.constant 0 : i1
```
Comment (durga4github) [inline mlir/test/Integration/GPU/CUDA/sm90/gemm_pred_f32_f16_f16_128x128x128.mlir:209]: well, I thought we will search replace all instances of this to %phase_c0 ;-)

Comment (grypp) [reply mlir/test/Integration/GPU/CUDA/sm90/gemm_pred_f32_f16_f16_128x128x128.mlir]: Right :) I fixed that. 
(happens when you are doing this in the end of your day)
